lib/stdlib/doc/src/queue.xml

```diff
@@ -44,9 +44,18 @@
       not lists. Improper lists cause internal crashes.
       An index out of range for a queue also causes
       a failure with reason <c>badarg</c>.</p>
+      <p><em>Example:</em></p>
+      <pre>
+1> <input>queue:len([]).</input>
+** exception error: bad argument</pre>
```

Comment:
This is IMO a bad example of invalid input. `len/1` expects an argument of type `queue`, which is an _opaque_ type. Thus, same as you can't assume anything about what it _is_, you can't make any assumptions about what it is _not_.

Reply:
Right, I was trying to show that if one passes the wrong argument, you get the described `badarg`.
If this is confusing or does not go well in here, I can move :)

Reply:
Maybe it would be better to just exemplify with another function, like giving an atom to `queue:from_list/1` or something, where the expected argument type is not opaque.